src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/JSonTest.java

```diff
@@ -0,0 +1,188 @@
+                .name(TEST_TABLE)
+                .addColumn(PK_COL, DbColumnType.LONG)
+                .addColumn(JSON_COL, DbColumnType.JSON)
+                .pkFields(PK_COL)
+                .build();
+        dbEngine.addEntity(testEntity);
+    }
+
+    /**
+     * Scenario for an insert using persist().
+     */
+    @Test
+    public void normalInsertTest() throws DatabaseFactoryException, DatabaseEngineException {
+        dbEngine.beginTransaction();
+        EntityEntry jsonTestEntry = getTestEntry();
+        dbEngine.persist(TEST_TABLE, jsonTestEntry, false);
+        dbEngine.commit();
+        checkInsertedValue(JSON_PART_1, JSON_PART_2);
+    }
+
+    /**
+     * Scenario for inserts in batch updates.
+     */
+    @Test
+    public void batchInsertTest() throws DatabaseFactoryException, DatabaseEngineException {
+        dbEngine.beginTransaction();
+        dbEngine.addBatch(TEST_TABLE, getTestEntry());
+        dbEngine.flush();
+        dbEngine.commit();
+        checkInsertedValue(JSON_PART_1, JSON_PART_2);
+    }
+
+    /**
+     * Scenario for an update of a JSON field using prepared statements
+     */
+    @Test
+    public void prepStmtUpdateTest() throws DatabaseEngineException, DatabaseFactoryException, NameAlreadyExistsException, ConnectionResetException {
+        // Insert entry first
+        normalInsertTest();
+
+        // Update JSON column
+        dbEngine.beginTransaction();
+        String PS_NAME = "MyPS";
+        Update upd = update(table(TEST_TABLE)).set(eq(column(JSON_COL), lit("?"))).where(eq(column(PK_COL), lit("?")));
+        dbEngine.createPreparedStatement(PS_NAME, upd);
+        dbEngine.clearParameters(PS_NAME);
+        dbEngine.setParameter(PS_NAME, 1, "{" + JSON_PART_1 + ", " + JSON_PART_3 + "}", DbColumnType.JSON);
+        dbEngine.setParameter(PS_NAME, 2, PK_VALUE);
+        dbEngine.executePS(PS_NAME);
+        dbEngine.commit();
+        checkInsertedValue(JSON_PART_1, JSON_PART_3);
+
+    }
+
+    /**
+     * Creates the test entry value used in most tests.
+     *
+     * @return
+     */
+    private EntityEntry getTestEntry() {
+        return entry()
+                .set(PK_COL, PK_VALUE)
+                .set(JSON_COL, JSON_VALUE)
+                .build();
+    }
+
+    /**
+     * Checks that the test table has a single entry and with the values as expected.
+     */
+    private void checkInsertedValue(String jsonPart1, String jsonPart2) throws DatabaseEngineException {
+        Expression query = select(all()).from(table(TEST_TABLE));
+        List<Map<String, ResultColumn>> results = dbEngine.query(query);
+        assertEquals("One value inserted", 1, results.size());
+        Map<String, ResultColumn> firstRow = results.get(0);
+        assertNotNull("Inserted row is not null", firstRow);
+        assertThat("JSon value is as expected", firstRow.get(JSON_COL).toString(), anyOf(
+                is("{" + jsonPart1 + ", " + jsonPart2 + "}"),
```

Comment:
This might fail on an engine that trims spaces when returning JSON. I would just trim them.